src/liquid/components/ld-input/ld-input.tsx

```diff
@@ -245,6 +255,16 @@ export class LdInput implements InnerFocusable, ClonesAttributes {
   }
 
   componentWillLoad() {
+    // Add readonly prop files.
+    Object.defineProperty(this.el, 'files', {
+      get: () => {
+        if (this.isInputTypeFile(this.input)) {
+          return this.input.files
+        }
+        return undefined
+      },
+    })
```

Comment:
`this.input` is not yet defined when the `componentWillLoad` hook is executed. Another downside is, that I would need to cast `this.input` to `(this.input as HTMLInputElement)` again, because we are in a different function scope, so Typescript kind of ignores the type guard from the outside scope. I'll leave this as it is.